apps/andi/lib/andi_web/validators/dataset_validator.ex

```diff
@@ -67,4 +86,25 @@ defmodule AndiWeb.DatasetValidator do
         false
     end
   end
+
+  # Handles preconverting datasets from structs to maps for comparison purposes
```

Comment:
I was going to ask why we're doing this, but this comment meant I didn't need to. Thanks!